scripts/tag-reconciler/tag-reconciler.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/cri-o/cri-o/internal/version"
+	"github.com/cri-o/cri-o/scripts/utils"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/release-sdk/git"
+	"sigs.k8s.io/release-utils/env"
+)
+
+func main() {
+	logrus.SetFormatter(&logrus.TextFormatter{DisableTimestamp: true})
+	if err := run(); err != nil {
+		logrus.Fatalf("Unable to run: %v", err)
+	}
+}
+
+func run() error {
+	if !env.IsSet(utils.GithubTokenEnvKey) {
+		return fmt.Errorf(
+			"environment variable %q is not set", utils.GithubTokenEnvKey,
+		)
+	}
+
+	remote := env.Default(utils.GitRemoteEnvKey, "origin")
+	logrus.Infof("Using repository remote: %s", remote)
+
+	org := env.Default(utils.OrgEnvKey, utils.CrioOrgRepo)
+	logrus.Infof("Using repository organization: %s", org)
+
+	if err := git.ConfigureGlobalDefaultUserAndEmail(); err != nil {
+		return fmt.Errorf("unable to configure global default user and email: %w", err)
+	}
+
+	repo, err := git.OpenRepo(".")
+	if err != nil {
+		return fmt.Errorf("unable to open local repository: %w", err)
+	}
+
+	for _, minorVersion := range version.ReleaseMinorVersions {
+		baseBranchName := utils.BranchPrefix + minorVersion // Creates "release-x.y" format.
+
+		sv, err := utils.GetCurrentVersionFromReleaseBranch(repo, baseBranchName) // Returns "x.y.z" format.
+		if err != nil {
+			return fmt.Errorf("current version from release branch %s to semver: %w", minorVersion, err)
+		}
+
+		currentReleaseVersion := utils.VersionPrefix + sv.String()
+		exists, err := doesCurrentReleaseVersionTagExist(repo, baseBranchName, currentReleaseVersion)
+		if err != nil {
+			return fmt.Errorf("current version from release branch %s to semver: %w", minorVersion, err)
+		}
+		if !exists {
+			if err := pushTagToRemote(repo, currentReleaseVersion, remote, baseBranchName); err != nil {
+				return fmt.Errorf("push tag %s: %w", currentReleaseVersion, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func pushTagToRemote(repo *git.Repo, tag, remote, releaseBranch string) error {
+	commitMessage := "version: bump to " + tag
+
+	logrus.Infof("Adding a new version tag: %s", tag)
+	if err := repo.Tag(tag, commitMessage); err != nil {
+		return fmt.Errorf("create tag %s: %w", tag, err)
+	}
+
+	logrus.Info("Pushing changes")
+	if err := repo.PushToRemote(remote, releaseBranch); err != nil {
+		return fmt.Errorf("pushing to remote: %s: %w", remote, err)
+	}
```

Comment:
That does not push the tag. We either need to use `PushToRemoteWithOptions` or run a `exec` command to push the tag explicitly.

Reply:
PushToRemoteWithOptions does not have an option to push tag or do you mean just adding the RemoteName and RemoteUrl?

https://pkg.go.dev/github.com/go-git/go-git/v5#PushOptions

Reply:
Can you just do an `git push origin tag <tag_name>` using the `os/exec` package? I think that's the safest way for now.